### PR TITLE
Added weight initialization in DCGAN

### PR DESCRIPTION
**Pull request by rutvij-25**
As weight initialization is also a part of DCGAN paper, I thought I'd add it to the code :)) #309

### CODE REVIEW

File: vision/dcgan_mnist/dcgan_mnist.jl [hunk 1]
```diff
@@ -30,12 +31,18 @@ function create_output_image(gen, fixed_noise, hparams)
     return image_array
 end
 
+  
+# weight initialization
+function random_normal(shape...)
+    return map(Float32,rand(Normal(0,0.02),shape...))
+end
```
Comment (darsnack) [inline vision/dcgan_mnist/dcgan_mnist.jl:38]:
```suggestion
dcgan_init(shape...) = randn(Float32, shape...) * 0.02
```

File: vision/dcgan_mnist/dcgan_mnist.jl [hunk 2]
```diff
@@ -5,6 +5,7 @@ using Flux.Losses: logitbinarycrossentropy
 using Images
 using MLDatasets
 using Statistics
+using Distributions:Normal
```
Comment (darsnack) [inline vision/dcgan_mnist/dcgan_mnist.jl:8]:
```suggestion
```

File: vision/dcgan_mnist/dcgan_mnist.jl [hunk 3]
```diff
@@ -30,12 +31,18 @@ function create_output_image(gen, fixed_noise, hparams)
...
+  
+# weight initialization
+function random_normal(shape...)
+    return map(Float32,rand(Normal(0,0.02),shape...))
+end
+
 function Discriminator()
     return Chain(
-            Conv((4, 4), 1 => 64; stride = 2, pad = 1),
+            Conv((4, 4), 1 => 64; stride = 2, pad = 1,init=random_normal),
```
Comment (darsnack) [inline vision/dcgan_mnist/dcgan_mnist.jl:42]: Match the formatting spacing here and elsewhere.

```suggestion
            Conv((4, 4), 1 => 64; stride = 2, pad = 1, init = random_normal),
```

**Review by darsnack — CHANGES_REQUESTED** [review]
Thanks for this! Can you include a comment with reference to the paper where the initialization procedure comes from?

File: vision/dcgan_mnist/dcgan_mnist.jl (hunk 2 above)
Comment (rutvij-25) [reply vision/dcgan_mnist/dcgan_mnist.jl:8]: Is it necessary to import Distributions?

File: vision/dcgan_mnist/dcgan_mnist.jl (hunk 1 above)
Comment (rutvij-25) [reply vision/dcgan_mnist/dcgan_mnist.jl:38]: Will do:)

File: vision/dcgan_mnist/dcgan_mnist.jl (hunk 2 above)
Comment (darsnack) [reply vision/dcgan_mnist/dcgan_mnist.jl]: I deleted the import because we aren't using `Normal` anymore. `randn` comes as part of Base.

**Comment by darsnack** [comment]
Is there a reason that you closed the PR?

File: vision/dcgan_mnist/dcgan_mnist.jl [hunk 4]
```diff
@@ -30,12 +30,16 @@ function create_output_image(gen, fixed_noise, hparams)
     return image_array
 end
 
+  
+# weight initialization as given in the "4.DETAILS OF ADVERSARIAL TRAINING" section of the paper
```
Comment (darsnack) [inline vision/dcgan_mnist/dcgan_mnist.jl:34]: Can you include a link to the paper (maybe on arXiv?)

**Review by darsnack — CHANGES_REQUESTED** [review]

**Comment by rutvij-25** [comment]
Hey @darsnack ,I am kinda new to github and I was thinking of making a new pull request with the required changes ,is there any other way?

File: vision/dcgan_mnist/dcgan_mnist.jl (hunk 4 above)
Comment (rutvij-25) [reply vision/dcgan_mnist/dcgan_mnist.jl:34]: Yes will do

**Comment by darsnack** [comment]
You can continue pushing to your branch. New commits will automatically be added onto this PR.

**Comment by rutvij-25** [comment]
I have done the changes

**Comment by darsnack** [comment]
Did you commit + push? Right now, there are no new changes.

**Review by darsnack — APPROVED** [review]
Great, thanks!

**Comment by rutvij-25** [comment]
Welcome! This was my first merged PR , hoping to contribute more to FLUX!

**Pull request merged**